napari/layers/labels/labels.py

```diff
@@ -996,13 +1003,14 @@ def _raw_to_displayed(self, raw, data_slice: Tuple[slice] = None):
             # Update the cache
             self._cached_labels[data_slice][update_mask] = labels_to_map
         else:
-            self._cached_labels = np.zeros_like(labels)
-            self._cached_mapped_labels = np.zeros_like(
-                labels, dtype=np.float32
-            )
-            self._cached_labels[data_slice] = sliced_labels.copy()
+            _cached_labels = np.zeros_like(labels)
+            _cached_mapped_labels = np.zeros_like(labels, dtype=np.float32)
+            _cached_labels[data_slice] = sliced_labels.copy()
             labels_to_map = sliced_labels
 
+            self._cached_labels = _cached_labels
+            self._cached_mapped_labels = _cached_mapped_labels
```

Comment:
Could you explain your reasoning behind this change?

Reply:
in this case, could you try casting it instead?

Reply:
The issue is with the slicing on the LHS of `self._cached_labels[data_slice] = sliced_labels.copy()`. I've simplified this a bit, but to do the slicing I think this still needs an intermediate variable.